articles/how-to-setup-nodejs-express-for-react/index.md

```diff
@@ -0,0 +1,179 @@
+
+Line 9 to 11 – It will set a `GET route` that we will, later on, fetch from our client-side React App.
+
+### Setting up the proxy
+In this step, the Webpack development server was auto-generated when we ran the `create-react-app` command.  Our react App is running on the Webpack development server on the frontend.  We can proxy the API requests from the client-side to the API on the server-side. The API  on the server-side(Express server) will be running on port 5000.
+
+First, set up the proxy to navigate into the client directory and locate its `package.json` file. Add the below line to it.
+
+"proxy": "http://localhost:5000"
+
+The modified `package.json` file will look as below:
+
+```typescript
+{
+  "name": "client",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@testing-library/jest-dom": "^5.11.10",
+    "@testing-library/react": "^11.2.6",
+    "@testing-library/user-event": "^12.8.3",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "react-scripts": "4.0.3",
+    "web-vitals": "^1.1.1"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject"
+  },
+  "proxy": "http://localhost:5000"
+}
+```
+
+The modified `package.json` file will let the webpack proxy the API requests to the Express backend server running on port 5000.
+
+### Calling Express backend server from React
+First is to browse into `client/src` and edit the file `App.js` to appear as below:
+
+```typescript
+import React, { Component } from 'react';
+import logo from './logo.svg';
+import './App.css';
+
+class App extends Component {
+state = {
+    data: null
+  };
+
+  componentDidMount() {
+    this.callBackendAPI()
+      .then(res => this.setState({ data: res.express }))
+      .catch(err => console.log(err));
+  }
+    // fetching the GET route from the Express server which matches the GET route from server.js
+  callBackendAPI = async () => {
+    const response = await fetch('/express_backend');
+    const body = await response.json();
+
+    if (response.status !== 200) {
+      throw Error(body.message) 
+    }
+    return body;
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <header className="App-header">
+          <img src={logo} className="App-logo" alt="logo" />
+          <h1 className="App-title">Welcome to React</h1>
+        </header>
+        <p className="App-intro">{this.state.data}</p>
+      </div>
+    );
+  }
+}
+
+export default App;
+```
+
+Inside of `componentDidMount()`, the function `callBackendAPI()` is called. This function will fetch the route previously created in the Express server and then `set{data: null}` to the fetch request-response.
+On the line that contains `this.state.data` renders new data  to the [DOM](https://eloquentjavascript.net/14_dom.html)
+
+### Running the App
+First will be to navigate to root directory `demoreactapp` and run the below command:
```

Comment:
Rewrite this sentence. We navigate to the root directory `demoreactapp` and run the command below: